tests/functional/s/super/superfluous_parens_walrus_py38.py

```diff
@@ -0,0 +1,86 @@
+"""Test the superfluous-parens warning with python 3.8 functionality (walrus operator)"""
+# pylint: disable=missing-function-docstring,invalid-name
+if not (x := False):
+    print(x)
+
+if not ((x := 1)):  # [superfluous-parens]
+    pass
+
+if not ((((x := 1)))):  # [superfluous-parens]
+    pass
+
+i = 1
+y = 1
+
+if odd := is_odd(i): # [undefined-variable]
+    pass
+if not (foo := 5):
+    pass
+
+if not ((x := y)):  # [superfluous-parens]
+    pass
+
+if ((x := y)):   # [superfluous-parens]
+    pass
+
+assert (ret := str(42))
+
+# False positive (str, tuple)
+A = [int(n) for n in (nums := '123')]
```

Comment:
A-R are not valid python code (you cannot use the walrus operator there).
I think we can therefore exclude them?

Reply:
Yeap, we can also remove Z :)